### PR TITLE
Allow compiling against slightly different tensorflow directory layouts.

### PR DESCRIPTION
**Pull request by pschuh**
Checks to see if the directory layout of --tensorflow-host-include-dir is 
third_party/tensorflow/... and then adjusts all the paths accordingly. When building against normal tensorflow, everything should work normally.

### CODE REVIEW

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/CMakeLists.txt [hunk 1]
```diff
@@ -21,16 +21,14 @@ endif()
...
-
 set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 set(swift_stdlib_compile_flags "${SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS}")
 list(APPEND swift_stdlib_compile_flags "-Xllvm" "-sil-inline-generics")
 list(APPEND swift_stdlib_compile_flags "-Xllvm" "-sil-partial-specialization")
 list(APPEND swift_stdlib_compile_flags "-Xfrontend" "-enable-sil-ownership")
 list(APPEND swift_stdlib_compile_flags "-swift-version" "4")
 list(APPEND swift_stdlib_compile_flags "-force-single-frontend-invocation")
-list(APPEND swift_stdlib_compile_flags "-Xcc" "-I${TF_INCLUDE_DIR}")
+# list(APPEND swift_stdlib_compile_flags "-Xcc" "-I${TF_INCLUDE_DIR}")
```
Comment (mhong) [inline stdlib/public/TensorFlow/CMakeLists.txt]: is this change intentional?

File: cmake/modules/FindTensorFlow.cmake [hunk 2]
```diff
@@ -4,8 +4,13 @@
 include(FindPackageHandleStandardArgs)
 
 find_path(TF_INCLUDE_DIR
-  NAMES tensorflow/c
+  NAMES third_party/tensorflow/c tensorflow/c
   HINTS ${SWIFT_TENSORFLOW_TARGET_INCLUDE_DIR} /usr/include /usr/local/include)
+if (EXISTS ${TF_INCLUDE_DIR}/third_party/tensorflow/c/c_api.h)
+  set(TF_PATH_ADJUSTMENT "third_party")
+else()
+  set(TF_PATH_ADJUSTMENT "")
```
Comment (mhong) [inline cmake/modules/FindTensorFlow.cmake]: consider adding a comment that the else case is the normal setup, for "mainstream" development, CI, etc.

The "if" case is experimental, and not covered by CI as intended.

File: stdlib/public/CTensorFlow/CMakeLists.txt [hunk 3]
```diff
@@ -53,20 +56,31 @@ foreach(sdk ${TARGET_SDKS})
...
+          "${copy_tf_headers_path_adjustment}")
 
       list(APPEND outputs "${output_dir}/${dst}")
     endmacro(copy_tf_headers)
 
-    copy_tf_headers("${TF_INCLUDE_DIR}/tensorflow/c/c_api.h" c_api.h)
-    copy_tf_headers("${TF_INCLUDE_DIR}/tensorflow/c/c_api_experimental.h" c_api_experimental.h)
-    copy_tf_headers("${TF_INCLUDE_DIR}/tensorflow/c/eager/c_api.h" c_api_eager.h)
+    if (TF_PATH_ADJUSTMENT)
+      set(adjusted_tf_include_dir "${TF_INCLUDE_DIR}/${TF_PATH_ADJUSTMENT}")
```
Comment (mhong) [inline stdlib/public/CTensorFlow/CMakeLists.txt]: should we make `adjusted_tf_include_dir` all capital, for consistency?

**Review by mhong — APPROVED** [review]
I believe this is an experimental change, and one use case is so that we can integrate a google-specific build of TF with OSS Swift. LGTM.

File: stdlib/public/CTensorFlow/CMakeLists.txt (hunk 3 above)
Comment (pschuh) [reply stdlib/public/CTensorFlow/CMakeLists.txt]: If you notice above, the "local" sets are lowercase. I believe due to this I'm being consistent. Correct me if I'm wrong.

File: stdlib/public/TensorFlow/CMakeLists.txt (hunk 1 above)
Comment (pschuh) [reply stdlib/public/TensorFlow/CMakeLists.txt]: I believe this is superceeded by the modulemap, and thus is not needed.

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
I'll ask cmake expert @dan-zheng take a look at this.

**Comment by pschuh** [comment]
Looks like it was not pressed.
